pkg/apiserver/rest/dualwriter_mode1.go

```diff
@@ -27,36 +33,193 @@ func NewDualWriterMode1(legacy LegacyStorage, storage Storage) *DualWriterMode1
 
 // Create overrides the behavior of the generic DualWriter and writes only to LegacyStorage.
 func (d *DualWriterMode1) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
-	ctx = klog.NewContext(ctx, d.Log)
-	return d.Legacy.Create(ctx, obj, createValidation, options)
+	log := d.Log.WithValues("kind", options.Kind)
+	ctx = klog.NewContext(ctx, log)
+	var method = "create"
+
+	startLegacy := time.Now().UTC()
+	res, err := d.Legacy.Create(ctx, obj, createValidation, options)
+	if err != nil {
+		log.Error(err, "unable to create object in legacy storage")
+		d.recordLegacyDuration(true, mode, options.Kind, method, startLegacy)
+		return res, err
+	}
+	d.recordLegacyDuration(false, mode, options.Kind, method, startLegacy)
+
+	accessorCreated, err := meta.Accessor(res)
+	if err != nil {
+		log.Error(err, "unable to get accessor for created object")
+	}
+
+	accessorOld, err := meta.Accessor(obj)
+	if err != nil {
+		log.Error(err, "unable to get accessor for old object")
+	}
+
+	enrichObject(accessorOld, accessorCreated)
+
+	go func() {
+		startStorage := time.Now().UTC()
+		ctx, _ := context.WithTimeoutCause(ctx, time.Second*10, errors.New("storage create timeout"))
+		_, err := d.Storage.Create(ctx, obj, createValidation, options)
+		defer d.recordStorageDuration(err != nil, mode, options.Kind, method, startStorage)
```

Comment:
nit: what's the point of the `defer` here?

Reply:
not really important, considering it's the last step anyway, it's more of an habit, but I'll remove it :)